src/httpd.c

```diff
@@ -389,65 +389,57 @@ serve_file(struct evhttp_request *req, const char *uri)
       return;
     }
 
-  ret = lstat(path, &sb);
-  if (ret < 0)
+
+  if (!realpath(path, deref))
     {
-      DPRINTF(E_WARN, L_HTTPD, "Could not lstat() %s: %s\n", path, strerror(errno));
+      DPRINTF(E_LOG, L_HTTPD, "Could not dereference %s: %s\n", path, strerror(errno));
 
       httpd_send_error(req, HTTP_NOTFOUND, "Not Found");
-
       return;
     }
 
-  if (S_ISLNK(sb.st_mode))
+  if (strlen(deref) >= PATH_MAX)
```

Comment:
I don't understand this check, since deref is PATH_MAX. So the only way strlen(deref) can be >= PATH_MAX is if the null terminator is after the buffer, but that would be a segmentation fault. Also, the realpath doc says "The resulting pathname is stored as a null-terminated string, up to a maximum of PATH_MAX bytes,". So sounds like no check is required?